test-clients/src/main/java/com/hedera/services/bdd/suites/token/UniqueTokenManagementSpecs.java

```diff
@@ -34,84 +34,104 @@
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.SUCCESS;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.TOKEN_WAS_DELETED;
 
 public class UniqueTokenManagementSpecs extends HapiApiSuite {
 	private static final org.apache.logging.log4j.Logger log = LogManager.getLogger(UniqueTokenManagementSpecs.class);
 	private static final String TOKEN_NAME = "token";
-
+	private static final String SUPPLY_KEY = "supplyKey";
 	public static void main(String... args) {
 		new UniqueTokenManagementSpecs().runSuiteSync();
 	}
 
 	@Override
 	protected List<HapiApiSpec> getSpecsInSuite() {
 		return List.of(new HapiApiSpec[]{
-						sadPathWithRepeatedMetadata(),
-						getTokenNftInfoWork(),
-						sadPathWithFrozenToken(),
+						getTokenNftInfoWorks(),
 						uniqueTokenHappyPath(),
-						happyPathOneMintFiveMetadata(),
-						happyPathFiveMintOneMetadata(),
+						failsWithFrozenToken(),
+						failsWithDeletedToken(),
+						failsWithRepeatedMetadata(),
+						happyPathWithBatchMetadata(),
+						happyPathMintMultipleWithIdenticalMetadata(),
 				}
 		);
 	}
 
+
 	private HapiApiSpec uniqueTokenHappyPath() {
 		return defaultHapiSpec("UniqueTokenHappyPath")
 				.given(
-						newKeyNamed("supplyKey"),
+						newKeyNamed(SUPPLY_KEY),
 						cryptoCreate(TOKEN_TREASURY),
 						tokenCreate(TOKEN_NAME)
 								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
 								.supplyType(TokenSupplyType.INFINITE)
 								.initialSupply(0)
-								.supplyKey("supplyKey")
+								.supplyKey(SUPPLY_KEY)
 								.treasury(TOKEN_TREASURY)
 				).when(
 						mintToken(TOKEN_NAME, List.of(ByteString.copyFromUtf8("memo"), ByteString.copyFromUtf8("memo1"))).via("mintTxn")
 				).then(
-
-						getAccountBalance(TOKEN_TREASURY)
-								.hasTokenBalance(TOKEN_NAME, 2),
-
 						getReceipt("mintTxn")
-								.hasPriorityStatus(ResponseCodeEnum.SUCCESS),
+								.hasPriorityStatus(SUCCESS)
+								.hasCostAnswerPrecheck(SUCCESS),
 
 						getTokenNftInfo(TOKEN_NAME, 1)
 								.hasSerialNum(1)
 								.hasMetadata(ByteString.copyFromUtf8("memo"))
-								.hasTokenID(TOKEN_NAME),
+								.hasTokenID(TOKEN_NAME)
+								.hasAccountID(TOKEN_TREASURY),
 
 						getTokenNftInfo(TOKEN_NAME, 2)
 								.hasSerialNum(2)
 								.hasMetadata(ByteString.copyFromUtf8("memo1"))
 								.hasTokenID(TOKEN_NAME)
+								.hasAccountID(TOKEN_TREASURY),
+
+						getTokenNftInfo(TOKEN_NAME, 3)
+								.hasCostAnswerPrecheck(INVALID_NFT_ID),
+
+						getAccountBalance(TOKEN_TREASURY)
+								.hasTokenBalance(TOKEN_NAME, 2),
+						getTokenInfo(TOKEN_NAME)
+								.hasTreasury(TOKEN_TREASURY)
+								.hasTotalSupply(0),
+
+						getAccountInfo(TOKEN_TREASURY)
+								.hasToken(relationshipWith(TOKEN_NAME))
 				);
 	}
 
-	private HapiApiSpec happyPathOneMintFiveMetadata() {
-		return defaultHapiSpec("happyPathOneMintFiveMetadata")
+	private HapiApiSpec happyPathWithBatchMetadata() {
```

Comment:
This test seems the same and is covered by `UniqueTokenHappyPath`. In both cases we are minting more 1 one nft.